tests/test_local_env.py

```diff
@@ -58,23 +57,62 @@ def test_determine_package_dir__various_paths_in_poetry2nix_env(
         }
     )
     path = tmp_path / subdir
-    expect = tmp_path / f"lib/python{major}.{minor}/site-packages"
-    assert LocalPackageResolver.determine_package_dir(path) == expect
+    expect = {tmp_path / f"lib/python{major}.{minor}/site-packages"}
+    assert set(LocalPackageResolver.find_package_dirs(path)) == expect
 
 
 @pytest.mark.parametrize(
     "subdir", [pytest.param(d, id=f"pep582:{d}") for d in pep582_subdirs]
 )
-def test_determine_package_dir__various_paths_in_pypackages(write_tmp_files, subdir):
+def test_find_package_dirs__various_paths_in_pypackages(write_tmp_files, subdir):
     # A directory structure that resembles a minimal PEP582 __pypackages__ dir:
     tmp_path = write_tmp_files(
         {
             f"__pypackages__/{major}.{minor}/lib/some_package.py": "",
         }
     )
     path = tmp_path / subdir
-    expect = tmp_path / f"__pypackages__/{major}.{minor}/lib"
-    assert LocalPackageResolver.determine_package_dir(path) == expect
+    expect = {tmp_path / f"__pypackages__/{major}.{minor}/lib"}
+    assert set(LocalPackageResolver.find_package_dirs(path)) == expect
+
+
+@pytest.mark.parametrize(
+    "subdir",
+    [pytest.param(d, id=f"pep582:{d}") for d in pep582_subdirs]
+    + [pytest.param(".venv/" + d, id=f"venv:.venv/{d}") for d in env_subdirs],
+)
+def test_find_package_dirs__envs_with_multiple_package_dirs(write_tmp_files, subdir):
+    # A directory structure that contains multiple Python environments, and
+    # multiple package dirs inside each Python environments:
+    tmp_path = write_tmp_files(
+        {
+            f"__pypackages__/{major}.{minor}/lib/first_package.py": "",
+            f"__pypackages__/{major}.{minor + 1}/lib/second_package.py": "",
+            ".venv/bin/python": "",
+            f".venv/lib/python{major}.{minor}/site-packages/third_package.py": "",
+            f".venv/lib/python{major}.{minor + 1}/site-packages/fourth_package.py": "",
+        }
+    )
+    path = tmp_path / subdir
+    actual = set(LocalPackageResolver.find_package_dirs(path))
+
+    def expected_package_dirs(base: Path, subdir: str) -> Iterator[Path]:
+        # When given:
+        # __pypackages__/  -->  return all .../X.Y/lib dirs
+        # __pypackages__/X.Y/*  -->  return the only matching .../lib dir
+        # .venv/  -->  return all .../lib/pythonX.Y/site-packages dirs
+        # .venv/lib/pythonX.Y/* --- return the only matching .../site-packages dir
+        if subdir.startswith("__pypackages__"):
+            yield base / f"__pypackages__/{major}.{minor}/lib"
+            if f"{major}.{minor}" not in subdir:
+                yield base / f"__pypackages__/{major}.{minor + 1}/lib"
+        else:
+            assert subdir.startswith(".venv/")
+            yield base / f".venv/lib/python{major}.{minor}/site-packages"
+            if f"{major}.{minor}" not in subdir:
+                yield base / f".venv/lib/python{major}.{minor + 1}/site-packages"
+
```

Comment:
```suggestion
        is_version_subdir = f"{major}.{minor}" not in subdir
        if subdir.startswith("__pypackages__"):
            yield base / f"__pypackages__/{major}.{minor}/lib"
            if is_version_subdir:
                yield base / f"__pypackages__/{major}.{minor + 1}/lib"
        elif subdir.startswith(".venv/"):
            yield base / f".venv/lib/python{major}.{minor}/site-packages"
            if is_version_subdir:
                yield base / f".venv/lib/python{major}.{minor + 1}/site-packages"
        else:
            raise ValueError("Unsupported subdir")

```

Reply:
I believe the `is_version_subdir` should instead be named `is_not_version_subdir` or maybe `is_version_agnostic`?

Reply:
`is_version_agnostic` ❤️

Reply:
Fixed in e0d7c83.